From cba92348129e956e0e8f77a3f2418cc80797e835 Mon Sep 17 00:00:00 2001
From: Eric Van Norman <ericvn@us.ibm.com>
Date: Wed, 13 Apr 2022 12:22:04 -0500
Subject: Test envoy update (#38332)

* Update to fix latest Envoy builds

* Run update_deps.sh

* Change from true to false

* Update goldens

* Additional updates
---
 istio.deps                                    |  2 +-
 pkg/bootstrap/config.go                       |  1 +
 pkg/bootstrap/testdata/all_golden.json        | 32 +++++++++----------
 pkg/bootstrap/testdata/auth_golden.json       | 12 +++----
 pkg/bootstrap/testdata/authsds_golden.json    | 12 +++----
 pkg/bootstrap/testdata/default_golden.json    | 12 +++----
 .../testdata/metrics_no_statsd_golden.json    | 24 +++++++-------
 pkg/bootstrap/testdata/running_golden.json    | 24 +++++++-------
 pkg/bootstrap/testdata/runningsds_golden.json | 24 +++++++-------
 .../testdata/stats_inclusion_golden.json      | 12 +++----
 .../testdata/tracing_datadog_golden.json      | 14 ++++----
 .../testdata/tracing_lightstep_golden.json    | 14 ++++----
 .../tracing_opencensusagent_golden.json       | 12 +++----
 .../testdata/tracing_stackdriver_golden.json  | 22 ++++++-------
 .../tracing_tls_custom_sni_golden.json        | 14 ++++----
 .../testdata/tracing_tls_golden.json          | 14 ++++----
 .../testdata/tracing_zipkin_golden.json       | 14 ++++----
 pkg/bootstrap/testdata/xdsproxy_golden.json   | 12 +++----
 pkg/envoy/instance_test.go                    |  2 +-
 .../testdata/envoy_bootstrap_v2.tmpl.json     |  5 ++-
 prow/release-commit.sh                        |  2 +-
 21 files changed, 142 insertions(+), 138 deletions(-)

diff --git a/istio.deps b/istio.deps
index 9c2be1c063..195b9e9d08 100644
--- a/istio.deps
+++ b/istio.deps
@@ -4,6 +4,6 @@
     "name": "PROXY_REPO_SHA",
     "repoName": "proxy",
     "file": "",
-    "lastStableSHA": "77b805d60bcfef0e0c879fd3f145bae77c02f76d"
+    "lastStableSHA": "430da1f7b354ab96218f314d79cb9dc57e6b3226"
   }
 ]
diff --git a/pkg/bootstrap/config.go b/pkg/bootstrap/config.go
index c41b5457a0..b3fbc9fe40 100644
--- a/pkg/bootstrap/config.go
+++ b/pkg/bootstrap/config.go
@@ -283,6 +283,7 @@ func extractRuntimeFlags(cfg *model.NodeMetaProxyConfig) map[string]string {
 		"envoy.reloadable_features.require_strict_1xx_and_204_response_headers":                                "false",
 		"re2.max_program_size.error_level":                                                                     "32768",
 		"envoy.reloadable_features.http_reject_path_with_fragment":                                             "false",
+		"envoy.reloadable_features.no_extension_lookup_by_name":                                                "false",
 	}
 	if !StripFragment {
 		// Note: the condition here is basically backwards. This was a mistake in the initial commit and cannot be reverted
diff --git a/pkg/bootstrap/testdata/all_golden.json b/pkg/bootstrap/testdata/all_golden.json
index e684bdb505..ddee403d04 100644
--- a/pkg/bootstrap/testdata/all_golden.json
+++ b/pkg/bootstrap/testdata/all_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -419,7 +419,7 @@
           }]
         }
       }
-      
+
       ,
       {
         "name": "envoy_metrics_service",
@@ -450,8 +450,8 @@
           }]
         }
       }
-      
-      
+
+
       ,
       {
         "name": "envoy_accesslog_service",
@@ -481,7 +481,7 @@
           }]
         }
       }
-      
+
     ],
     "listeners":[
       {
@@ -598,10 +598,10 @@
       }
     }
   }
-  
+
   ,
   "stats_sinks": [
-    
+
     {
       "name": "envoy.stat_sinks.metrics_service",
       "typed_config": {
@@ -614,11 +614,11 @@
         }
       }
     }
-    
-    
+
+
     ,
-    
-    
+
+
     {
       "name": "envoy.stat_sinks.statsd",
       "typed_config": {
@@ -628,15 +628,15 @@
         }
       }
     }
-    
+
   ]
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/auth_golden.json b/pkg/bootstrap/testdata/auth_golden.json
index ebfba2172d..9a2becead6 100644
--- a/pkg/bootstrap/testdata/auth_golden.json
+++ b/pkg/bootstrap/testdata/auth_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -500,13 +500,13 @@
       }
     ]
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/authsds_golden.json b/pkg/bootstrap/testdata/authsds_golden.json
index 889b9ffbb9..9fcba6495e 100644
--- a/pkg/bootstrap/testdata/authsds_golden.json
+++ b/pkg/bootstrap/testdata/authsds_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -500,13 +500,13 @@
       }
     ]
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/default_golden.json b/pkg/bootstrap/testdata/default_golden.json
index 226637177b..80e09869b5 100644
--- a/pkg/bootstrap/testdata/default_golden.json
+++ b/pkg/bootstrap/testdata/default_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -500,13 +500,13 @@
       }
     ]
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/metrics_no_statsd_golden.json b/pkg/bootstrap/testdata/metrics_no_statsd_golden.json
index 5fc479e317..739bf4368d 100644
--- a/pkg/bootstrap/testdata/metrics_no_statsd_golden.json
+++ b/pkg/bootstrap/testdata/metrics_no_statsd_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "envoy_metrics_service",
@@ -427,8 +427,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -531,10 +531,10 @@
       }
     ]
   }
-  
+
   ,
   "stats_sinks": [
-    
+
     {
       "name": "envoy.stat_sinks.metrics_service",
       "typed_config": {
@@ -547,17 +547,17 @@
         }
       }
     }
-    
-    
-    
+
+
+
   ]
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/running_golden.json b/pkg/bootstrap/testdata/running_golden.json
index 0fc025ef55..aaf8b038cf 100644
--- a/pkg/bootstrap/testdata/running_golden.json
+++ b/pkg/bootstrap/testdata/running_golden.json
@@ -15,7 +15,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -401,7 +401,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -424,8 +424,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -542,12 +542,12 @@
       }
     }
   }
-  
+
   ,
   "stats_sinks": [
-    
-    
-    
+
+
+
     {
       "name": "envoy.stat_sinks.statsd",
       "typed_config": {
@@ -557,15 +557,15 @@
         }
       }
     }
-    
+
   ]
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/runningsds_golden.json b/pkg/bootstrap/testdata/runningsds_golden.json
index 830116435e..6f10a55924 100644
--- a/pkg/bootstrap/testdata/runningsds_golden.json
+++ b/pkg/bootstrap/testdata/runningsds_golden.json
@@ -15,7 +15,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -401,7 +401,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -424,8 +424,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -542,12 +542,12 @@
       }
     }
   }
-  
+
   ,
   "stats_sinks": [
-    
-    
-    
+
+
+
     {
       "name": "envoy.stat_sinks.statsd",
       "typed_config": {
@@ -557,15 +557,15 @@
         }
       }
     }
-    
+
   ]
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/stats_inclusion_golden.json b/pkg/bootstrap/testdata/stats_inclusion_golden.json
index 6c9c454314..6c18abbbca 100644
--- a/pkg/bootstrap/testdata/stats_inclusion_golden.json
+++ b/pkg/bootstrap/testdata/stats_inclusion_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -489,8 +489,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -593,13 +593,13 @@
       }
     ]
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_datadog_golden.json b/pkg/bootstrap/testdata/tracing_datadog_golden.json
index 5b248c131f..0d0e2c1fc8 100644
--- a/pkg/bootstrap/testdata/tracing_datadog_golden.json
+++ b/pkg/bootstrap/testdata/tracing_datadog_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "datadog_agent",
@@ -418,8 +418,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -533,13 +533,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_lightstep_golden.json b/pkg/bootstrap/testdata/tracing_lightstep_golden.json
index 36274a4dc7..16901805b8 100644
--- a/pkg/bootstrap/testdata/tracing_lightstep_golden.json
+++ b/pkg/bootstrap/testdata/tracing_lightstep_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "lightstep",
@@ -426,8 +426,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -541,13 +541,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_opencensusagent_golden.json b/pkg/bootstrap/testdata/tracing_opencensusagent_golden.json
index 4953bab252..1906b7be54 100644
--- a/pkg/bootstrap/testdata/tracing_opencensusagent_golden.json
+++ b/pkg/bootstrap/testdata/tracing_opencensusagent_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -522,13 +522,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_stackdriver_golden.json b/pkg/bootstrap/testdata/tracing_stackdriver_golden.json
index 812e8348e8..3c15baa1f8 100644
--- a/pkg/bootstrap/testdata/tracing_stackdriver_golden.json
+++ b/pkg/bootstrap/testdata/tracing_stackdriver_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -508,7 +508,7 @@
       "@type": "type.googleapis.com/envoy.config.trace.v3.OpenCensusConfig",
       "stackdriver_exporter_enabled": true,
       "stackdriver_project_id": "my-sd-project",
-      
+
       "stackdriver_grpc_service": {
         "google_grpc": {
           "target_uri": "cloudtrace.googleapis.com",
@@ -526,15 +526,15 @@
           }]
         },
         "initial_metadata": [
-        
+
           {
             "key": "x-goog-user-project",
             "value": "my-sd-project"
           }
-        
+
         ]
       },
-      
+
       "stdout_exporter_enabled": true,
       "incoming_trace_context": ["CLOUD_TRACE_CONTEXT", "TRACE_CONTEXT", "GRPC_TRACE_BIN", "B3"],
       "outgoing_trace_context": ["CLOUD_TRACE_CONTEXT", "TRACE_CONTEXT", "GRPC_TRACE_BIN", "B3"],
@@ -549,14 +549,14 @@
        }
      }
   }}
-  
-  
-  
+
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_tls_custom_sni_golden.json b/pkg/bootstrap/testdata/tracing_tls_custom_sni_golden.json
index 51710fe113..9b071047b7 100644
--- a/pkg/bootstrap/testdata/tracing_tls_custom_sni_golden.json
+++ b/pkg/bootstrap/testdata/tracing_tls_custom_sni_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -420,8 +420,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -538,13 +538,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_tls_golden.json b/pkg/bootstrap/testdata/tracing_tls_golden.json
index a0cd7f6758..44d87329d2 100644
--- a/pkg/bootstrap/testdata/tracing_tls_golden.json
+++ b/pkg/bootstrap/testdata/tracing_tls_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -420,8 +420,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -538,13 +538,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/tracing_zipkin_golden.json b/pkg/bootstrap/testdata/tracing_zipkin_golden.json
index fe4d41e36b..2fcbed40e0 100644
--- a/pkg/bootstrap/testdata/tracing_zipkin_golden.json
+++ b/pkg/bootstrap/testdata/tracing_zipkin_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,7 +396,7 @@
           }
         }
       }
-      
+
       ,
       {
         "name": "zipkin",
@@ -419,8 +419,8 @@
           }]
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -537,13 +537,13 @@
       }
     }
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/bootstrap/testdata/xdsproxy_golden.json b/pkg/bootstrap/testdata/xdsproxy_golden.json
index ceb91a08c9..c194785f30 100644
--- a/pkg/bootstrap/testdata/xdsproxy_golden.json
+++ b/pkg/bootstrap/testdata/xdsproxy_golden.json
@@ -10,7 +10,7 @@
       "layers": [
           {
             "name": "global config",
-            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
+            "static_layer": {"envoy.deprecated_features:envoy.config.listener.v3.Listener.hidden_envoy_deprecated_use_original_dst":"true","envoy.reloadable_features.no_extension_lookup_by_name":"false","envoy.reloadable_features.http_reject_path_with_fragment":"false","envoy.reloadable_features.require_strict_1xx_and_204_response_headers":"false","overload.global_downstream_max_connections":"2147483647","re2.max_program_size.error_level":"32768"}
           },
           {
               "name": "admin",
@@ -396,8 +396,8 @@
           }
         }
       }
-      
-      
+
+
     ],
     "listeners":[
       {
@@ -500,13 +500,13 @@
       }
     ]
   }
-  
-  
+
+
   ,
   "cluster_manager": {
     "outlier_detection": {
       "event_log_path": "/dev/stdout"
     }
   }
-  
+
 }
diff --git a/pkg/envoy/instance_test.go b/pkg/envoy/instance_test.go
index 942e3e402b..f40d63d17f 100644
--- a/pkg/envoy/instance_test.go
+++ b/pkg/envoy/instance_test.go
@@ -32,7 +32,7 @@
 	"istio.io/istio/pkg/test/util/tmpl"
 )
 
-var envoyLogFormat = envoy.LogFormat("[ENVOY][%Y-%m-%d %T.%e][%t][%l][%n]")
+var envoyLogFormat = envoy.LogFormat("[ENVOY][%Y-%m-%d %T.%e][%t][%l][%n] %v")
 
 func TestNewWithoutConfigShouldFail(t *testing.T) {
 	g := NewWithT(t)
diff --git a/pkg/envoy/testdata/envoy_bootstrap_v2.tmpl.json b/pkg/envoy/testdata/envoy_bootstrap_v2.tmpl.json
index b6ebadcf66..4dd417516d 100644
--- a/pkg/envoy/testdata/envoy_bootstrap_v2.tmpl.json
+++ b/pkg/envoy/testdata/envoy_bootstrap_v2.tmpl.json
@@ -75,7 +75,10 @@
                     ]
                   },
                   "http_filters": [{
-                    "name": "envoy.filters.http.router"
+                    "name": "envoy.filters.http.router",
+                    "typed_config": {
+                      "@type": "type.googleapis.com/envoy.extensions.filters.http.router.v3.Router"
+                    }
                   }]
                 }
               }
diff --git a/prow/release-commit.sh b/prow/release-commit.sh
index 137f61f12b..b7cf109e55 100755
--- a/prow/release-commit.sh
+++ b/prow/release-commit.sh
@@ -32,7 +32,7 @@ DOCKER_HUB=${DOCKER_HUB:-gcr.io/istio-testing}
 GCS_BUCKET=${GCS_BUCKET:-istio-build/dev}
 
 # Use a pinned version in case breaking changes are needed
-BUILDER_SHA=2d930dc6ef98e9ce45a8723e3f5fb2f76dfd9ebc
+BUILDER_SHA=5a7ecb6dc03df10bbb2919d31bfa566017a21a9d
 
 # Reference to the next minor version of Istio
 # This will create a version like 1.4-alpha.sha
-- 
2.35.3

